### ISSUE TITLE
Multiple line append at end of line

### ISSUE THREAD

**Issue by davidbrochart** [jonathanslenders/ptvim#108]
When going to the end of a line (`$`), then `Ctrl-v` and select several lines (`j`), then `Shift-a` to append, the text is inserted at the position corresponding to the end of the first line instead of being inserted at the end of every line.

**Comment by davidbrochart** [comment]
I realize it was already the case before prompt-toolkit 2.0.

**Comment by jonathanslenders** [comment]
Good point! I noticed that if you press `$` and then move up/down, the cursor also remains at the end of the lines. There are many operations where this `$` takes effect that we have to support.

BTW: if you find other bugs related to text objects or cursor movement; you can report the issues in the prompt_toolkit repository.

**Comment by davidbrochart** [comment]
That's tricky though because you will have to differentiate between being at the last character of a line and being at the end of a line, which shows the same cursor position on the screen...
